src/composables/useTresContextProvider/index.ts

```diff
@@ -8,6 +8,26 @@ import { useCamera } from '../useCamera'
 import type { UseRendererOptions } from '../useRenderer'
 import { useRenderer } from '../useRenderer'
 import { extend } from '../../core/catalogue'
+import { useLogger } from '../useLogger'
+
+export interface InternalState {
```

Comment:
@andretchen0 if the name `renderModeState` is meant to add clarity that the property isn't just a static setting but potentially a reactive state that could change over time, then we should rename most of the properties inside the `TresContext` type to state, like `controlsState` and `rendererState` etc.

Imo being inside of the `TresContext` which is already a state implies its nature without the need to add extra verbosity.

Reply:
Hey @alvarosabu 

I'd be fine with dropping `State` off the end, as most variables are some sort of state. (But then the name `renderMode` will collide with the other `renderMode`.)

The current name – `internalState` – doesn't say anything about what the object is used for. Since it's about `renderMode`-stuff, I thought that'd make it more obvious to future readers.

But no problem leaving it as is.

Reply:
Sounds good!